x/operator/keeper/abci.go

```diff
@@ -9,35 +9,61 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+type OperatorUSDValue struct {
+	Staking                 sdkmath.LegacyDec
+	SelfStaking             sdkmath.LegacyDec
+	StakingAndWaitUnbonding sdkmath.LegacyDec
+}
+
 // CalculateUSDValueForOperator calculates the total and self usd value for the
 // operator according to the input assets filter and prices.
 func (k *Keeper) CalculateUSDValueForOperator(
 	ctx sdk.Context,
+	isForSlash bool,
 	operator string,
 	assetsFilter map[string]interface{},
 	decimals map[string]uint32,
 	prices map[string]operatortypes.Price,
-) (sdkmath.LegacyDec, sdkmath.LegacyDec, error) {
-	usdValue := sdkmath.LegacyNewDec(0)
-	selfUSDValue := sdkmath.LegacyNewDec(0)
+) (OperatorUSDValue, error) {
+	var err error
+	ret := OperatorUSDValue{
+		Staking:                 sdkmath.LegacyNewDec(0),
+		SelfStaking:             sdkmath.LegacyNewDec(0),
+		StakingAndWaitUnbonding: sdkmath.LegacyNewDec(0),
+	}
 	// iterate all assets owned by the operator to calculate its voting power
 	opFuncToIterateAssets := func(assetID string, state *assetstypes.OperatorAssetInfo) error {
-		price := prices[assetID]
-		decimal := decimals[assetID]
-		usdValue = usdValue.Add(CalculateUSDValue(state.TotalAmount, price.Value, decimal, price.Decimal))
-		// calculate the token amount from the share for the operator
-		selfAmount, err := delegationkeeper.TokensFromShares(state.OperatorShare, state.TotalShare, state.TotalAmount)
-		if err != nil {
-			return err
+		var price operatortypes.Price
+		var decimal uint32
+		if isForSlash {
```

Comment:
Can you add a comment here to explain why the price is fetched again?

Reply:
okay

Reply:
Sorry, this is not clear. Why is it null and not passed? Is there a specific reasoning behind this difference?